vamb/__main__.py

```diff
@@ -1442,7 +1447,41 @@ def run_bin_default(opt: BinDefaultOptions):
         FastaOutput.try_from_common(opt.common),
         None,
     )
-    del latent
+
+
+class RunDefault:
+    pass
+
+
+class RunTrain:
+    pass
+
+
+class RunCluster:
+    def __init__(self, latent: Path):
+        self.latent = latent
```

Comment:
@sgalkina I suggested this pattern, to mimick an ADT (or `enum`, in Rust parlance). My idea is to 1) statically check that the argument is the correct type, and 2) make it easier to catch errors if a new partial mode is added. Is there a better way in Python to make ADTs that can be caught by the type checker?